tests/typechecker/function_pass_by_reference.jakt

```diff
@@ -0,0 +1,16 @@
+/// Expect:
+/// - output: "2\n"
+
+function add(i: i64, adder: &function(anon i: i64) -> i64) -> i64  {
+    return adder(i)
+}
+
+function main() {
+    let adder = function(anon i: i64) -> i64 {
+        return i + 1
+    }
+
+    let a = add(i: 1, adder: &adder)
```

Comment:
now that we have https://github.com/SerenityOS/jakt/pull/1115, can this be `let a = add(i: 1, &adder)`?

Reply:
Yeah, I have updated that now.

Reply:
Just to note I have also updated the PR to allow for checking for reference types in `check_types_for_compat`